src/trio/_core/_run.py

```diff
@@ -1820,6 +1821,11 @@ async def python_wrapper(orig_coro: Awaitable[RetT]) -> RetT:
         return task
 
     def task_exited(self, task: Task, outcome: Outcome[Any]) -> None:
+        if task in GLOBAL_PARKING_LOT_BREAKER:
+            for lot in GLOBAL_PARKING_LOT_BREAKER[task]:
+                lot.break_lot(task)
```

Comment:
Maybe add a line comment here like `# break parking lots associated with a task that exited` because otherwise the warning (from multiple tasks breaking a lot) might point to here in the stack trace.

Reply:
Sorry to be specific:
```suggestion
                lot.break_lot(task)  # task exited while it was a parking lot breaker
```
Because this line specifically will be highlighted in the default traceback handler. (even though custom ones can get context)

Reply:
I started playing around with this... and it appears that the `RuntimeWarning` can get transformed into a TrioInternalError >.>

Reply:
we might also want to raise an error if trying to register a task as a breaker that has already exited

Reply:
> Nevermind, I was completely incorrect about how warnings seem. I'm concerned that this might cause a weird warning location if someone adds multiple tasks as breakers for a single lot and they all exit, but I'm not sure how to better account for that.

yeah it currently is giving me zero useful location info on where the task exited. And even trying to play around with `stacklevel` and [`skip_file_prefixes` from 3.12](https://docs.python.org/3/library/warnings.html#warnings.warn) I fail to get any change in behavior. Not sure if that's me failing or trio doing sus things

Reply:
At worst we could make the warning into a no-op when in trio internals (just duplicate the checks above this).

... but that's not very useful.

Reply:
Maybe this is enough justification to make `broken_by` a list? It's a slightly worse API but I think it's fine.